### ISSUE TITLE
Adding custom field result in not being able to find columns from table

### ISSUE THREAD

**Issue by BenjaminGK** [metabase/metabase#12397]
**Describe the bug**
Adding custom column result in errors showing that a field doesn't exist though it does in fact exist and error only shows when trying to add a custom field like =CONCAT([Field1],[Field2])

**Logs**
<img width="1920" alt="Screen Shot 2563-04-25 at 19 44 05" src="https://user-images.githubusercontent.com/23311430/80280264-2dce6300-872d-11ea-9c18-de2768a5ce7e.png">

**To Reproduce**
1. Go to 'Custom Question'
2. Click on 'Custom Field'
3. Click Preview

**Expected behavior**

**Screenshots**

**Information about your Metabase Installation:**

```
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.122 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.5+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.5",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.5+10",
    "os.name": "Linux",
    "os.version": "4.15.0-96-generic",
    "user.language": "en",
    "user.timezone": "Europe/Stockholm"
  },
  "metabase-info": {
    "databases": [
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "5.7.29-0ubuntu0.16.04.1-log"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.5.1"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-04-21",
      "tag": "v0.35.3",
      "branch": "release-0.35.x",
      "hash": "1d424cb"
    },
    "settings": {
      "report-timezone": "Europe/Berlin"
    }
  }
}
```
**Severity**
Difficult to specify as some tables work while others doesn't

**Additional context**

**Comment by flamber** [comment]
Hi @BenjaminGK 
I might be misunderstanding, but where's the error?
What are you doing between step 2 and 3? Can you provide steps to reproduce with Sample Dataset?
Is this a duplicate of #12268?
Those console `compile error Error: "Unknown Field: coun"` are log messages, not warnings or errors. It will make an "error" for every character you type that doesn't match a field or function.

**Comment by BenjaminGK** [comment]
@flamber Sorry about the missing information. I am not able to replicate it with the sample DB but I have added another screenshot that would hopefully help.
<img width="1616" alt="Screen Shot 2563-04-25 at 20 54 59" src="https://user-images.githubusercontent.com/23311430/80281792-e1d4eb80-8737-11ea-92d9-4d4f772d4362.png">

**Comment by flamber** [comment]
@BenjaminGK 
I have a feeling that your password manager is messing with something (or whatever the green circle to the right is). And because you have a column called something with `payment`, while the Sample Dataset doesn't, is probably why you can only reproduce it on this `companies` table.
There isn't much we can do about that, since there's no standard for telling those extensions/addons to bugger off.

**Comment by BenjaminGK** [comment]
@flamber 
I had another look in another browser without any extension and I got the same issue.
What I found out is that it happens for tables with many columns.
Looking at the SQL that is being used the system created it as a sub-query like this:
```
SELECT
source.field1 AS field1,
source.field2 AS field2,
source.Customcolumn AS Customcolumn,
companies.payment AS payment
FROM (SELECT concat(field1, '1') AS Customcolumn, companies.field1 AS field1, companies.field2 AS field2, companies.payment FROM companies) source
```

I have removed some of the columns just show, it works as long as I remove a bunch of columns but If I include all or most of them it will generate the SQL seen above. As you can see it looks for the payment column in companies but that table no longer exist based on the sub-query that has been renamed.
Hope this maybe helps and someone else are able to replicate this as well

**Comment by sbelak** [comment]
If it only happens with tables with a lot of columns, it's very likely same as ##12304 & fixed with #12387

**Comment by sbelak** [comment]
Fixed in #12387
